backtester/eventhandlers/strategies/strategies.go

```diff
@@ -31,39 +31,29 @@ func LoadStrategyByName(name string, useSimultaneousProcessing bool) (Handler, e
 
 func createNewStrategy(name string, useSimultaneousProcessing bool, h Handler) (Handler, error) {
 	if h == nil {
-		return nil, fmt.Errorf("cannot load %v supported strategies contains %w", name, common.ErrNilPointer)
+		return nil, fmt.Errorf("cannot load strategy %q: %w", name, common.ErrNilPointer)
 	}
+
 	if !strings.EqualFold(name, h.Name()) {
 		return nil, nil
 	}
-	// create new instance so strategy is not shared across all tasks
+
 	strategyValue := reflect.ValueOf(h)
-	if strategyValue.IsNil() {
-		return nil, fmt.Errorf("cannot load %v supported strategies element is a %w", name, common.ErrNilPointer)
-	}
-	strategyElement := strategyValue.Elem()
-	if !strategyElement.IsValid() {
-		return nil, fmt.Errorf("cannot load %v strategy element is invalid %w", name, common.ErrTypeAssertFailure)
-	}
-	strategyType := strategyElement.Type()
-	if strategyType == nil {
-		return nil, fmt.Errorf("cannot load %v strategy type is a %w", name, common.ErrNilPointer)
-	}
-	newStrategy := reflect.New(strategyType)
-	if newStrategy.IsNil() {
-		return nil, fmt.Errorf("cannot load %v new instance of strategy is a %w", name, common.ErrNilPointer)
+	if strategyValue.Kind() != reflect.Ptr || strategyValue.IsNil() {
+		return nil, fmt.Errorf("cannot load strategy %q: handler must be a non-nil pointer, got %T", name, h)
 	}
-	strategyInterface := newStrategy.Interface()
-	if strategyInterface == nil {
-		return nil, fmt.Errorf("cannot load %v new instance of strategy is not an interface. %w", name, common.ErrTypeAssertFailure)
-	}
-	strategy, ok := strategyInterface.(Handler)
+
+	// create new instance so strategy is not shared across all tasks
+	strategy, ok := reflect.New(strategyValue.Elem().Type()).Interface().(Handler)
```

Comment:
🥃  _Aside:_ Feels like h should have been a func returning a Handler 🙂 
`NewSellEverything() Handler`. That's a bigger change and OOS.